src/loaders/genome_collection/compute_genome_attribs.py

```diff
@@ -309,38 +330,70 @@ def _run_microtrait(genome_id, fna_file, batch_dir):
+        library(microtrait)
+        extract_traits <- function(genome_file, out_dir) {
+          genome_file <- file.path(genome_file)
+          microtrait_result <- extract.traits(in_file = genome_file, out_dir = out_dir)
+          return(microtrait_result)
+        }
+        """
+        r_func = robjects.r(r_script)
+        r_result = r_func(fna_file, genome_dir)
+
+        microtrait_result = _get_r_list_element(r_result, 'microtrait_result')
+
+        trait_counts_atgranularity3 = _get_r_list_element(microtrait_result, 'trait_counts_atgranularity3')
+        trait_counts_atgranularity3_df = _r_table_to_df(trait_counts_atgranularity3)
+
+        trait_counts_atgranularity3_df.to_csv(os.path.join(genome_dir, 'trait_counts_atgranularity3.csv'), index=False)
+
+    except Exception as e:
+        raise ValueError(f'Error running microtrait on {fna_file}') from e
+    finally:
+        # remove the unpacked assembly file to save space
+        if remove_gz_file:
+            print(f'removing {fna_file}')
+            os.remove(fna_file)
+
+    selected_cols = ['microtrait_trait-value', 'microtrait_trait-displaynameshort', 'microtrait_trait-displaynamelong']
+    cell_info_df = trait_counts_atgranularity3_df[selected_cols]
+
+    # Extract the substring of the 'microtrait_trait-displaynamelong' column before the last colon character
+    # and assign it to a new 'category' column in the DataFrame
+    cell_info_df['category'] = cell_info_df['microtrait_trait-displaynamelong'].str.rsplit(':', 1).str[0]
+    # Use the microtrait_trait-displaynameshort column as the column name, microtrait_trait-displaynamelong column
+    # as the column description, and microtrait_trait-value as the cell value
+    cell_info_df = cell_info_df.rename(columns={'microtrait_trait-displaynameshort': 'col_name',
+                                                'microtrait_trait-displaynamelong': 'col_description',
+                                                'microtrait_trait-value': 'cell_val'})
+
+    categories, cells, celid, min_value, max_value = list(), list(), 0, float('inf'), float('-inf')
+    for category_name, category_df in cell_info_df.groupby('category'):
+        columns = list()
+        for i, row in category_df.iterrows():
+            columns.append(ColumnInformation(id=str(i), name=row['col_name'], description=row['col_description']))
+            cell_val = float(row['cell_val'])
+            cells.append(Cell(celid=str(celid), colid=str(i), val=cell_val))
+            min_value, max_value = min(min_value, cell_val), max(max_value, cell_val)
+            celid += 1
+        categories.append(ColumnCategory(category=category_name, columns=columns))
+
+    meta = HeatMapMeta(categories=categories, min_value=min_value, max_value=max_value)
+    row = HeatMapRow(kbase_id=genome_id, cells=cells)
+
+    with open(os.path.join(genome_dir, 'heatmap_meta.json'), "w") as heatmap_meta:
+        json.dump(dict(meta), heatmap_meta)
```

Comment:
is it possible that the heat map meta might difffer from microtrait run to microtrait run (other than the min and max values)? If so, how will you deal with that?

Reply:
they will be different? within the same load_ver?

Reply:
I don't know, that's why I'm asking

Reply:
I really don't think we need to worry about this.

Reply:
What I'm wondering about is whether if there's no trait found for a particular genome that column will be missing, that kind o fthing

Reply:
hmmmm. the genome is missing? I think what microtrait does is on the gene level. Do you mean the complete `row` or `meta` is empty? Even that in the result json will still be something like {kbase_id: 'foo', load_ver: 'bar', ... , categories: [], cells:[]}.

Reply:
No, a trait is missing. So different microtrait run outputs might have different or out of order trait lists

Reply:
I know see what you are asking. I will handle it the parsing script and keep a global variable to check it.

Reply:
:+1: